docs/book/src/developer/providers/migrations/v1.7-to-v1.8.md

```diff
@@ -0,0 +1,26 @@
+# Cluster API v1.7 compared to v1.8
+
+This document provides an overview over relevant changes between Cluster API v1.7 and v1.8 for
+maintainers of providers and consumers of our Go API.
+
+## Go version
+
+- The Go version used by Cluster API is Go 1.22.x
+
+## Dependencies
+
+**Note**: Only the most relevant dependencies are listed, `k8s.io/` and `ginkgo`/`gomega` dependencies in Cluster API are kept in sync with the versions used by `sigs.k8s.io/controller-runtime`.
```

Comment:
Let's drop the dependency section. Doesn't make much sense anymore now that we generate a dependency diff as part of the release notes